python/python/lancedb/db.py

```diff
@@ -338,9 +339,27 @@ def __init__(
             uri = uri.expanduser().absolute()
             Path(uri).mkdir(parents=True, exist_ok=True)
         self._uri = str(uri)
-
         self._entered = False
         self.read_consistency_interval = read_consistency_interval
+        self.storage_options = storage_options
+
+        if read_consistency_interval is not None:
+            read_consistency_interval_secs = read_consistency_interval.total_seconds()
+        else:
+            read_consistency_interval_secs = None
+
+        async def do_connect():
+            return await lancedb_connect(
+                sanitize_uri(uri),
+                None,
+                None,
+                None,
+                read_consistency_interval_secs,
+                None,
+                storage_options,
+            )
+
+        self._conn = AsyncConnection(LOOP.run(do_connect()))
```

Comment:
I don't think the wrapper does much. Can just directly pass the Future/Coroutine to `LOOP.run()`.
```suggestion
        do_connect = lancedb_connect(
            sanitize_uri(uri),
            None,
            None,
            None,
            read_consistency_interval_secs,
            None,
            storage_options,
        )

        self._conn = AsyncConnection(LOOP.run(do_connect))
```

Reply:
It seems like if we do that, we get an error when we call lancedb_connect:
```
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "/Users/albertlockett/Development/lancedb/python/main.py", line 3, in <module>
    db = connect(
        # "~/Desktop/lance_datasets/db05",
    ...<5 lines>...
        }
    )
  File "/Users/albertlockett/Development/lancedb/python/python/lancedb/__init__.py", line 118, in connect
    return LanceDBConnection(
        uri,
        read_consistency_interval=read_consistency_interval,
        storage_options=storage_options,
    )
  File "/Users/albertlockett/Development/lancedb/python/python/lancedb/db.py", line 361, in __init__
    do_connect = lancedb_connect(
        sanitize_uri(uri),
    ...<5 lines>...
        storage_options,
    )
RuntimeError: no running event loop
```
Wrapping the call in this function seems to relieve the error, which is why it's here. If you know an cleaner way to avoid this error, happy to try fix this up.

Reply:
Ah okay. Nevermind. There's probably some async runtime magic going on I don't understand 🤷